Source/CesiumRuntime/Private/CesiumEncodedMetadataUtility.cpp

```diff
@@ -0,0 +1,707 @@
+    ECesiumMetadataTrueType trueType =
+        UCesiumMetadataPropertyBlueprintLibrary::GetTrueType(property);
+    bool isArray = trueType == ECesiumMetadataTrueType::Array;
+    bool isNormalized =
+        UCesiumMetadataPropertyBlueprintLibrary::IsNormalized(property);
+
+    int64 componentCount;
+    if (isArray) {
+      trueType = UCesiumMetadataPropertyBlueprintLibrary::GetTrueComponentType(
+          property);
+      componentCount =
+          UCesiumMetadataPropertyBlueprintLibrary::GetComponentCount(property);
+    } else {
+      componentCount = 1;
+    }
+
+    int32 expectedComponentCount = 1;
+    switch (pExpectedProperty->Type) {
+    // case ECesiumPropertyType::Scalar:
+    //  expectedComponentCount = 1;
+    //  break;
+    case ECesiumPropertyType::Vec2:
+      expectedComponentCount = 2;
+      break;
+    case ECesiumPropertyType::Vec3:
+      expectedComponentCount = 3;
+      break;
+    case ECesiumPropertyType::Vec4:
+      expectedComponentCount = 4;
+    };
+
+    if (expectedComponentCount != componentCount) {
+      continue;
+    }
+
+    // Coerce the true type into the expected gpu component type.
+    ECesiumMetadataPackedGpuType gpuType = ECesiumMetadataPackedGpuType::None;
+    if (pExpectedProperty->ComponentType ==
+        ECesiumPropertyComponentType::Uint8) {
+      gpuType = ECesiumMetadataPackedGpuType::Uint8;
+    } else /*if (expected type is float)*/ {
+      gpuType = ECesiumMetadataPackedGpuType::Float;
+    }
+
+    if (pExpectedProperty->Normalized != isNormalized) {
+      continue;
+    }
+
+    // Only support normalization of uint8 for now
+    if (isNormalized && trueType != ECesiumMetadataTrueType::Uint8) {
+      continue;
+    }
+
+    EncodedPixelFormat encodedFormat =
+        getPixelFormat(gpuType, componentCount, isNormalized);
+
+    if (encodedFormat.format == EPixelFormat::PF_Unknown) {
+      continue;
+    }
+
+    CESIUM_TRACE(TCHAR_TO_UTF8(
+        *("Encode Property Array: " + featureTableDescription.Name)));
+
+    EncodedMetadataProperty& encodedProperty =
+        encodedFeatureTable.encodedProperties.Emplace_GetRef();
+    encodedProperty.name =
+        "FTB_" + featureTableDescription.Name + "_" + pair.Key;
+
+    int64 propertyArraySize = featureCount * encodedFormat.pixelSize;
+    encodedProperty.pTexture = MakeUnique<LoadedTextureResult>();
+    encodedProperty.pTexture->pTextureData =
+        createTexturePlatformData(propertyArraySize, 1, encodedFormat.format);
+
+    encodedProperty.pTexture->addressX = TextureAddress::TA_Clamp;
+    encodedProperty.pTexture->addressY = TextureAddress::TA_Clamp;
+    encodedProperty.pTexture->filter = TextureFilter::TF_Nearest;
+
+    if (!encodedProperty.pTexture->pTextureData) {
+      // TODO: print error?
+      break;
+    }
+
+    FTexture2DMipMap* pMip = new FTexture2DMipMap();
+    encodedProperty.pTexture->pTextureData->Mips.Add(pMip);
+    pMip->SizeX = propertyArraySize;
+    pMip->SizeY = 1;
```

Comment:
@kring That's a great point! Fortunately implementing something like that shouldn't change the encoding code at all since the written buffer will still be linear. It will change what we set the mip->SizeX and mip->SizeY to and also how we retrieve the property in the auto-generated material (i.e., the propertyArray.Load). 

What would be a good max row / max column size (in pixels)?